### DISCUSSION TITLE
Constant stream of "Connection was closed" log messages

### DISCUSSION THREAD

**Discussion by brianmay** [teslamate-org/teslamate#1280] [General]
Seems to happen at least a minute. Is this normal? Car is in "falling asleep" state.

```
2021-02-02 08:28:10.660 [info] Stream.Timeout / 9
2021-02-02 08:28:10.856 [info] Connection was closed (a:1|t:10|d:4). Reconnecting …
```

**Comment by joshentu** [comment]
Same issue here, even with the latest v1.21.2, missed few drives, tried sign out and sign in again, restart docker, but no luck

**Reply by brianmay** [reply]
For the missing events issue (no idea if that is related or not to the above messages), see #1274.

**Reply by joshentu** [reply]
Very similar log and error code as yours, not sure if its a bug or not, strangely everything works fine today even it shows "Scheduling token refresh in 1 h"

**Comment by joshentu** [comment]
2021-02-02 19:54:55.353 [error] GenServer TeslaMate.Api terminating
teslamate_1  | ** (MatchError) no match of right hand side value: {:error, %TeslaApi.Error{env: nil, message: "socket closed", reason: :token_refresh}}
teslamate_1  |     (teslamate 1.21.2) lib/teslamate/api.ex:150: TeslaMate.Api.handle_info/2
teslamate_1  |     (stdlib 3.14) gen_server.erl:689: :gen_server.try_dispatch/4
teslamate_1  |     (stdlib 3.14) gen_server.erl:765: :gen_server.handle_msg/6
teslamate_1  |     (stdlib 3.14) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Last message: :refresh_auth
teslamate_1  | 2021-02-02 19:54:57.148 [warn] Token refresh failed: %TeslaApi.Error{env: nil, message: "socket closed", reason: :token_refresh}
teslamate_1  | 2021-02-02 19:54:57.148 [info] Scheduling token refresh in 1 h
teslamate_1  | 2021-02-02 19:58:16.875 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …

**Reply by brianmay** [reply]
If that was with v1.21.2, looks like a bug to me in the latest token refreshing code.

Not seeing it though myself. Yet...

**Reply by joshentu** [reply]
As suggested by Adrian in another post, 

> delete the entry from the database table tokens and restart the application.

but n00b to docker so not sure exactly how to do it, will dig a bit further

**Reply by brianmay** [reply]
I am guessing the procedure will be something like (not tested):

```
$ psql --host $host --user teslamate teslamate
Password for user teslamate: 
psql (13.1 (Debian 13.1-1+b1), server 13.0)
Type "help" for help.

teslamate=> TRUNCATE tokens;
```
Not sure if you need to restart Teslamate for it to notice the change or not.

**Reply by joshentu** [reply]
Yes, found it in another disussion with

> docker-compose exec database psql -U teslamate -d teslamate
DELETE FROM tokens;

Restart docker and Teslamate, but now i am stuck in the sign in page with error of 

> Error: :authorization_failed
